File: DatingApp/Helper/UserParams.cs
namespace DatingApp.Helper
{
    public class UserParams :PaginationParams
    {
        public string CurrentUsername { get; set; }
        public int UserId { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 90;
        public string OrderBy { get; set; }
        public bool Likees { get; set; } = false;
        public bool Likers { get; set; } = false;
    }
}

File: ClientApp/src/app/shared/params/user-params.ts
import { IUser } from "../models/iUser";
import { QueryParams } from "./query-params";

export class UserParams implements QueryParams {
    pageNumber = 1;
    pageSize = 5;
    search: string;
    gender: string;
    minAge = 18;
    maxAge = 99;
    orderBy = 'lastActive';

    constructor(user: IUser) {
        this.gender = user.gender === 'female' ? 'Male' : 'Female';
    }
}